src/qml/filters/stabilize/ui.qml

```diff
@@ -42,6 +46,12 @@ Item {
         }
     }
 
+    function analyzeValueChanged() {
+        button.enabled = true
+        status.text = qsTr('Analysis required')
+        filter.set("results", '')
```

Comment:
Now, after analysis is complete and you leave the filter and return to it, it says Analysis required. That is because the option values are loaded when the UI loads, which triggers this function and clears the results. Clearing results really does make analysis required. If you remove this line there is stil a race condition that typically causes this function to be called and give the superficial impression that it is. The solution is to make a top-level Component.onCompleted script block that controls the order of setting option values, the button state, and label. I did that and also refactored some things as well.